packages/client/src/declarations/selectors.ts

```diff
@@ -28,12 +28,16 @@ export const getInitialDeclarationsLoaded = (
   getKey(store, 'initialDeclarationsLoaded')
 
 const selectDeclaration =
-  <T extends IDeclaration | undefined>(declarationId: string) =>
+  <T extends IDeclaration | undefined>(declarationId?: string) =>
   (store: IStoreState) =>
-    getKey(store, 'declarations').find(({ id }) => declarationId === id) as T
+    declarationId
+      ? (getKey(store, 'declarations').find(
+          ({ id }) => declarationId === id
+        ) as T)
+      : (undefined as T)
 
 export const useDeclaration = <T extends IDeclaration | undefined>(
-  declarationId: string
+  declarationId?: string
```

Comment:
Good question! I think you are right, it doesn't.

The code this uses looked something like this:
```
getKey(store, 'declarations').find(({ id }) => declarationId === id) as T
```

declaration was received mostly from params, and the declarationId was never parsed before this stage, so I took the liberty of passing the undefined on till the bitter end. We can change it of course, since it is not optimal at all.

Reply:
yea I get it, we really need to introduce some validation of sorts for route params